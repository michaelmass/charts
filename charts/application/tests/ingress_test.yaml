suite: test ingress
templates:
  - 'ingress.yaml'
tests:
  - it: manifest should match snapshot
    set:
      ingress:
        enabled: true
    asserts:
      - matchSnapshot: {}
  - it: labels & annotations should be set
    set:
      common:
        labels:
          test_common: test
        annotations:
          test_common: test
      ingress:
        enabled: true
        labels:
          test_ingress: bar
        annotations:
          test_ingress: bar
    asserts:
      - isKind:
          of: Ingress
      - isSubset:
          path: metadata.labels
          content:
            test_common: test
            test_ingress: bar
      - isSubset:
          path: metadata.annotations
          content:
            test_common: test
            test_ingress: bar
