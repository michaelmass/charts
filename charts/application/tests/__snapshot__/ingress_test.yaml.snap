manifest should match snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: application
        helm.sh/chart: application-0.0.1
      name: RELEASE-NAME-application
      namespace: NAMESPACE
    spec:
      rules:
        - host: application.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-application
                    port:
                      name: http
                path: /
                pathType: ImplementationSpecific
